modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AMQPConstructStore.java

```diff
@@ -122,15 +122,26 @@ public void addQueue(AndesQueue queue, boolean isLocal) throws AndesException {
      * remove a queue
      *
      * @param queueName name of the queue to be removed
-     * @param isLocal   is this a local change
      * @throws AndesException
      */
-    public void removeQueue(String queueName, boolean isLocal) throws AndesException {
-        if (isLocal) {
-            andesContextStore.deleteQueueInformation(queueName);
-            //create the space created to keep message counter on this queue
-            messageStore.removeQueue(queueName);
-        }
+    public void removeLocalQueue(String queueName) throws AndesException {
+
+        // Remove the queue from internal maps
+        removeClusterQueue(queueName);
```

Comment:
I think it's better to invalidate the cache after removing the entry from DB, to avoid any race conditions. Since the reads can happen concurrently this can happen in a loaded environment.